modules/custom/openy_upgrade_tool/src/OpenyUpgradeLogManager.php

```diff
@@ -0,0 +1,588 @@
+    ModuleInstallerInterface $module_installer,
+    ThemeHandlerInterface $theme_handler,
+    LoggerChannelInterface $loggerChannel) {
+
+    $this->featuresManager = $features_manager;
+    $this->logger = $loggerChannel;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->configStorage = $config_storage;
+    $this->renderer = $renderer;
+    // Services necessary for \Drupal\Core\Config\ConfigImporter.
+    $this->eventDispatcher = $event_dispatcher;
+    $this->configManager = $config_manager;
+    $this->configFactory = $config_factory;
+    $this->lock = $lock;
+    $this->typedConfigManager = $typed_config;
+    $this->moduleHandler = $module_handler;
+    $this->moduleInstaller = $module_installer;
+    $this->themeHandler = $theme_handler;
+  }
+
+  /**
+   * Get entity storage that used for upgrade tool.
+   */
+  public function getLoggerEntityStorage() {
+    // TODO: Delete this, after deprecated 'logger_entity' deleting
+    // and create instance in __construct.
+    if ($this->loggerEntityStorage) {
+      return $this->loggerEntityStorage;
+    }
+
+    $this->loggerEntityStorage = $this->entityTypeManager
+      ->getStorage($this->getLoggerEntityTypeName());
+    return $this->loggerEntityStorage;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getOpenyConfigList() {
+    $features_configs = $this->featuresManager->listExistingConfig(TRUE);
+    // Get openy configs from features configs list.
+    $openy_configs = array_filter($features_configs, function ($module, $config) {
+      return strpos($module, 'openy') !== FALSE;
+    }, ARRAY_FILTER_USE_BOTH);
+    return array_keys($openy_configs);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isForceMode() {
+    return (bool) $this->configFactory
+      ->get('openy_upgrade_tool.settings')
+      ->get('force_mode');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function saveLoggerEntity($name, array $data, $message = NULL) {
+    try {
+      if ($this->getLoggerEntityTypeName() == 'logger_entity') {
+        // TODO: Delete this, 'logger_entity' is deprecated.
+        // Load logger entity with this config name.
+        $entities = $this->getLoggerEntityStorage()->loadByProperties([
+          'type' => 'openy_config_upgrade_logs',
+          'name' => $name,
+        ]);
+        if (empty($entities)) {
+          // Create new logger entity for this config name if not exist.
+          $logger_entity = $this->getLoggerEntityStorage()->create([
+            'type' => 'openy_config_upgrade_logs',
+          ]);
+        }
+        else {
+          $logger_entity = array_shift($entities);
+        }
+        $logger_entity->setName($name);
+        $logger_entity->setData([$name]);
+        $logger_entity->save();
+        return $logger_entity->id();
+      }
+      else {
+        // Load OpenyUpgradeLog entity with this config name.
+        $entities = $this->getLoggerEntityStorage()->loadByProperties([
```

Comment:
we have to limil load slicing by 100, memory saving wise

Reply:
Here we have limit by 'name' => $name, as result we can get only 1 or 0 entities. Multiple entities with the same name cannot exist